### DISCUSSION TITLE
Charting Library for FuncUI

### DISCUSSION THREAD

**Discussion by matthewcrews** [fsprojects/Avalonia.FuncUI#336] [Q&A]
Is there a charting library that is recommended for working with FuncUI? I specifically use the Elmish style of the loop.

**Comment by JaggerJo** [comment]
here are multiple ways to render charts in FuncUI.

- Use any [Avalonia Charting library](https://github.com/AvaloniaCommunity/awesome-avalonia#charts--plots--diagrams) and [write the bindings](https://funcui.avaloniaui.net/view-basics/how-to-create-bindings) you need for them.
- Use [Microcharts](https://github.com/microcharts-dotnet/Microcharts) and [render to an Avalonia Control](https://github.com/AvaloniaCommunity/Avalonia.Microcharts/blob/master/src/Avalonia.Microcharts/MicrochartControl.cs)
- Use [Plotly](https://plotly.net/) or any other library than can render charts as an image. Display the image.
- Use [Plotly](https://plotly.net/) and host the resulting chart HTML in an [embedded browser](https://github.com/OutSystems/CefGlue).
